### PR TITLE
Add initial SSL implementation

### PR DESCRIPTION
**Pull request by bassosimone**
This is very rough. In particular, we do not validate certificates, so
a man in the middle is very possible for now.

But I feel like it's important to have the basic scaffolding such
that things that should go in parallel over SSL can.

<strike>Crossing fingers as for Valgrind. I recall that used to reported
several uninitialized variables in OpenSSL.</strike>

### CODE REVIEW

**Comment by bassosimone** [comment]
- [ ] review and merge pull request
- [ ] add "bug" issue saying that we should validate certificates

**Comment by bassosimone** [comment]
Closes #485.

File: test/net/transport.cpp [hunk 1]
```diff
@@ -47,6 +47,20 @@ TEST_CASE("net::connect() can connect to open port") {
...
 
+TEST_CASE("net::connect() can connect to SSL port") {
+    if (CheckConnectivity::is_down()) {
+        return;
+    }
+    set_verbose(1);
+    loop_with_initial_event([]() {
+        connect("nexa.polito.it", 443, [](Error error, Var<Transport> txp) {
+            REQUIRE(!error);
+            txp->close([]() { break_loop(); });
```
Comment (AntonioLangiu) [inline test/net/transport.cpp]: :+1:

File: src/net/ssl-context.hpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#define SRC_NET_SSL_CONTEXT_HPP
+
+// Forward declarations
+struct ssl_st;
+struct ssl_ctx_st;
+
+namespace mk {
+namespace net {
+
+class SslContext {
```
Comment (AntonioLangiu) [inline src/net/ssl-context.hpp]: this should be not movable and not copyable.
we don't want to have copy of the singleton

Comment (bassosimone) [inline src/net/ssl-context.hpp]: I don't think you cannot easily get an instance of the class, given that the constructor is private, but I agree that for robustness we should declare it like so! Hold on, cooking the diff...

**Comment by AntonioLangiu** [comment]
Ok, i tested it locally with OSX and with ubuntu trusty on vagrant (using valgrind). 
It's ok for me!

**Comment by bassosimone** [comment]
Thanks for reviewing!

**Pull request merged**